sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -396,6 +396,25 @@ object MapGroups {
       UnresolvedDeserializer(encoderFor[T].deserializer, dataAttributes),
       groupingAttributes,
       dataAttributes,
+      Seq.empty,
```

Comment:
shall we just add a new parameter `dataOrder: Seq[SortOrder]` with default value Nil?

Reply:
I think that breaks binary compatibility with Spark 3.x

Reply:
You mean like
```
  def apply[K : Encoder, T : Encoder, U : Encoder](
      func: (K, Iterator[T]) => TraversableOnce[U],
      groupingAttributes: Seq[Attribute],
      dataAttributes: Seq[Attribute],
      child: LogicalPlan,
      dataOrder: Seq[SortOrder] = Nil): LogicalPlan = {
```
Looks unexpected that `child` is not the last argument. And argument order would deviate from `case class MapGroups`.

Reply:
It's internal API, I don't think binary compatibility matters here. We can just add the `dataOrder: Seq[SortOrder]` parameter before `child: LogicalPlan` and update existing code.